### PR TITLE
LPD-25757 | master

### PR DESCRIPTION
**Pull request by rachaelkoestartyo**
https://liferay.atlassian.net/browse/LPD-25757

### CODE REVIEW

**Comment by rachaelkoestartyo** [comment]
ci:test:sf

**Comment by rachaelkoestartyo** [comment]
ci:test:analytics-cloud

File: modules/test/playwright/tests/osb-faro-web/page.spec.ts [hunk 1]
```diff
@@ -128,3 +128,76 @@ test('shows outside pages in path analysis', async ({apiHelpers, page}) => {
...
+
+	await syncAnalyticsCloud(page, channelName);
+
+	await page.waitForTimeout(10000);
+
+	await page.getByText(pageTitle1).first().click();
+
+	await page.getByText(pageTitle2).first().click();
+
+	await page.waitForTimeout(10000);
```
Comment (ViniLopes20) [inline modules/test/playwright/tests/osb-faro-web/page.spec.ts:151]: Hey @rachaelkoestartyo, I think this pause should be placed after you generate the interaction with the page to guarantee that the event was sent. I believe this pause after the connection is not necessary. What do you think?

```suggestion
	await page.getByText(pageTitle1).first().click();

	await page.waitForTimeout(10000);

	await page.getByText(pageTitle2).first().click();

	await page.waitForTimeout(10000);
```

File: modules/test/playwright/tests/osb-faro-web/page.spec.ts [hunk 2]
```diff
@@ -128,3 +128,76 @@ test('shows outside pages in path analysis', async ({apiHelpers, page}) => {
...
+
+	const sitePage2 = await createSitePage(apiHelpers, pageTitle2);
+
+	const channelName = 'My Property - ' + getRandomString();
+
+	await syncAnalyticsCloud(page, channelName);
+
+	await page.waitForTimeout(10000);
+
+	await page.getByText(pageTitle1).first().click();
```
Comment (ViniLopes20) [inline modules/test/playwright/tests/osb-faro-web/page.spec.ts:147]: Hey @rachaelkoestartyo, This navigation doesn't appear correct to me. Here you are trying to find the element with the name of your created page and trying to click to access the page, but you are currently on the AC connection page with DXP and this element with your page does not appear in the UI.

My suggestion is that you create this function below. On that you can pass to the function the name of the page that you want to access (site name also if you had another site created) and your page will be loaded directly in the URL as a direct access to the page that was passed.

After that, you can use this `page.getByText` function to access page 2 as a reference page. What do you think?


```
export async function navigateToSitePage(page, siteName, pageName) {
    // Replace spaces with hyphens and convert to lowercase
    let pageNameURL = pageName.replace(/ /g, "-").toLowerCase();

    if (siteName) {
        // Replace spaces with hyphens and convert to lowercase
        let siteNameURL = siteName.replace(/ /g, "-").toLowerCase();

        // Construct URL and navigate, considering that a new site is already created in the instance
        await page.goto(`${liferayConfig.environment.baseUrl}/web/${siteNameURL}/${pageNameURL}`);
    } else {
        // Construct URL and navigate, considering the site is the default for Liferay DXP
        await page.goto(`${liferayConfig.environment.baseUrl}/${pageNameURL}`);
    }
}
```

File: modules/test/playwright/tests/osb-faro-web/page.spec.ts [hunk 3]
```diff
@@ -128,3 +128,76 @@ test('shows outside pages in path analysis', async ({apiHelpers, page}) => {
...
+		timeout: 100 * 1000,
+	});
+
+	await expect(page.getByText('1', {exact: true}).nth(1)).toBeVisible({
+		timeout: 100 * 1000,
+	});
+
+	await expect(page.getByText('1', {exact: true}).nth(2)).toBeVisible({
+		timeout: 100 * 1000,
+	});
```
Comment (ViniLopes20) [inline modules/test/playwright/tests/osb-faro-web/page.spec.ts:196]: Hey @rachaelkoestartyo, I think these should be the pages you should check to see if they appear in Path Analysis. What do you think?

**Steps:**
1. Go to My Page 1 (Using the `navigateToSitePage` function) - Direct access
2. Go to My Page 2 (Using `page.getByText`) - Exit Page
3. Go to My Page 1 (Using `page.getByText`) - Referral Page
4. Go to AC > Go to Pages > Path Analysis of My Page 1
5. Check that in the Referral column you have direct access and My Page 2 as referral pages
6. Check that in the Exit Page column has Drop Offs and My Page 2 as exit pages

**Review by ViniLopes20 — CHANGES_REQUESTED** [review]
Hey @rachaelkoestartyo, Thank you for working on this automation. Could you look at my comments?

**Comment by rachaelkoestartyo** [comment]
ci:test:analytics-cloud

File: modules/test/playwright/tests/osb-faro-web/page.spec.ts [hunk 4]
```diff
@@ -58,6 +58,20 @@ const goToWithReferrer = async function (page, referrer, url) {
...
+	if (siteName) {
+		let siteNameURL = siteName.replace(/ /g, "-").toLowerCase();
+
+		await page.goto(
+			`${liferayConfig.environment.baseUrl}/web/${siteNameURL}/` +
+				`${pageNameURL}`);
+	} else {
+		await page.goto(`${liferayConfig.environment.baseUrl}/${pageNameURL}`);
+	}
+};
```
Comment (ViniLopes20) [inline modules/test/playwright/tests/osb-faro-web/page.spec.ts:73]: Hey @rachaelkoestartyo, could you move this function to the `analyticsSettings.ts` file? This is a file that we put the functions that will be more reusable on the DXP side

**Review by ViniLopes20 — CHANGES_REQUESTED** [review]
Good job, @rachaelkoestartyo. Could you look at my comments?

**Comment by ViniLopes20** [comment]
Hey @rachaelkoestartyo, I noticed that your [test execution is failing in CI](https://testray.liferay.com/reports/production/logs/2024-05/test-1-35/test-portal-acceptance-pullrequest(master)/5804/playwright-js-tomcat90-mysql57-jdk8/1/0/playwright-report/index.html?authuser=0). You can follow these [steps](https://liferay.atlassian.net/wiki/spaces/ENGAC/pages/2895183873/How+to+run+the+Playwright+or+and+Poshi+tests+in+the+CI#%3Aci%3A-%3Aplaywright%3A-How-to-run-a-specific-Playwright-test-in-the-CI%3F) to run only your new automation in CI without having to run the AC suite completely

**Comment by rachaelkoestartyo** [comment]
ci:test:analytics-cloud

**Comment by rachaelkoestartyo** [comment]
ci:test:analytics-cloud

**Comment by rachaelkoestartyo** [comment]
ci:test:analytics-cloud

**Review by ViniLopes20 — APPROVED** [review]
Hi @rachaelkoestartyo, Thanks for working on it! 🎉

The code looks good to me!

**Comment by rachaelkoestartyo** [comment]
ci:test:sf

**Comment by rachaelkoestartyo** [comment]
ci:test:sf

**Comment by rachaelkoestartyo** [comment]
ci:test:sf

**Comment by rachaelkoestartyo** [comment]
ci:test:sf

**Comment by rachaelkoestartyo** [comment]
ci:test:sf

**Comment by rachaelkoestartyo** [comment]
ci:forward

**Pull request closed without merging**